srcpkgs/libgccjit/template

```diff
@@ -0,0 +1,241 @@
+# Template file for 'libgccjit'
+_majorver=10
+_minorver=${_majorver}.2
+_patchver=${_minorver}.1
+
+pkgname=libgccjit
+# we are using a 10-stable git snapshot alpine is using in order
+# to get regression fixes not yet incorporate into a stable release
+# it should be possible to switch back to stable with 10.3 or 11
+version=${_patchver}pre1
+wrksrc=gcc-${version/pre/_pre}
+revision=2
+short_desc="GNU Compiler Collection"
+maintainer="Enno Boland <gottox@voidlinux.org>"
+homepage="http://gcc.gnu.org"
+license="GFDL-1.2-or-later, GPL-3.0-or-later, LGPL-2.1-or-later"
+# *-musl builders have issues fetching https://ftp.gnu.org/gnu/mpc/mpc-1.1.0.tar.gz
+distfiles="https://dev.alpinelinux.org/~nenolod/gcc-${version/pre/_pre}.tar.xz"
+checksum="772cfd5d30eb9cc5a996fec9ba8cdcb45d37df1c9b5770610103b814b732c590"
+nopie=yes
+lib32disabled=yes
+
+# MIPS is untested and does not have go in crosstoolchains
+# 32-bit PowerPC seems weirdly broken since the 10.x release
+# 32-bit little endian PowerPC has no support at all (no GOARCH)
+case "$XBPS_TARGET_MACHINE" in
+	ppc64*) ;;
+	mips*-musl|ppc*) _have_gccgo=no ;;
+esac
+
+if [ "$CHROOT_READY" ]; then
+	hostmakedepends="tar texinfo perl flex"
+fi
+
+makedepends="zlib-devel gmp-devel mpfr-devel libmpc-devel 
+  libstdc++-devel-${version}_${revision} libssp-devel-${version}_${revision}"
```

Comment:
Both of these should already be available... And 

- you don't want to depend on a specific revision of a package from another template, since they can get updated independently from each other
- you shouldn't specify versions in `hostmakedepends` or `depends`. The versions that will be used will be the ones listed in the templates for each package, so they don't affect anything besides possibly breaking a build.

Reply:
so as it turns out this was due to my choice to substitute the source distfiles for these libraries with the void repo-provided packages. this causes problems when I attempt to cross-compile, so I reverted to the original distfiles def that pulls the source packages for mpfr, mpc, etc.

zlib-devel however had to remain included in order to compile, mainly because of the use of --with-system-zlib flag. If it's advisable to try to see if dropping this flag works, I can go ahead and test that.

Reply:
I think that @ericonr just means that `libstdc++-devel-${version}_${revision} libssp-devel-${version}_${revision}` ... i think the rest can stay

Reply:
gotcha, thanks @fosslinux. if you want to take a look at my comments below, I had to reduce this list just to include zlib-devel and rely on the original distfiles definition from gcc.